src/views/v2/list/top-bar/datepicker.php

```diff
@@ -14,25 +14,39 @@
  * @var string $today Today date in the `Y-m-d` format.
  *
  */
+use Tribe__Date_Utils as Dates;
+
+/**
+ * @todo @bordoni @lucatume Abstract the handling of these dates into a method/function
+ */
 $default_start_date = 'now';
 $selected_start_date_value = $this->get( [ 'bar', 'date' ], $default_start_date );
 if ( empty( $selected_start_date_value ) ) {
-	$selected_start_date_value = $default_start_date;
+	$first_event = $this->get( 'view' )->get_repository()->first();
+
```

Comment:
Maybe assign `$default_start_date` to `$selected_start_date_value` and have only the `if` to check if it's intance of and replace, to avoid the else?

Reply:
Just to be consistent with what we use elsewhere.

Reply:
On this case we need that else, because of the way initial `$selected_start_date_value` is set.
I wanted to avoid that else too.